src/THREE.MeshLine.js

```diff
@@ -1,782 +1,713 @@
 ;(function() {
-
-"use strict";
-
-var root = this
-
-var has_require = typeof require !== 'undefined'
-
-var THREE = root.THREE || has_require && require('three')
-if( !THREE )
-	throw new Error( 'MeshLine requires three.js' )
-
-function MeshLine() {
-
-	this.positions = [];
-
-	this.previous = [];
-	this.next = [];
-	this.side = [];
-	this.width = [];
-	this.indices_array = [];
-	this.uvs = [];
-	this.counters = [];
-	this.geometry = new THREE.BufferGeometry();
-
-	this.widthCallback = null;
-
-	// Used to raycast
-	this.matrixWorld = new THREE.Matrix4();
-}
-
-MeshLine.prototype.setMatrixWorld = function(matrixWorld) {
-	this.matrixWorld = matrixWorld;
-}
-
-
-MeshLine.prototype.setGeometry = function( g, c ) {
-	
-	this.widthCallback = c;
-
-	this.positions = [];
-	this.counters = [];
-	// g.computeBoundingBox();
-	// g.computeBoundingSphere();
-
-	// set the normals
-	// g.computeVertexNormals();
-	if( g instanceof THREE.Geometry ) {
-		for( var j = 0; j < g.vertices.length; j++ ) {
-			var v = g.vertices[ j ];
-			var c = j/g.vertices.length;
-			this.positions.push( v.x, v.y, v.z );
-			this.positions.push( v.x, v.y, v.z );
-			this.counters.push(c);
-			this.counters.push(c);
-		}
-	}
-
-	if( g instanceof THREE.BufferGeometry ) {
-		// read attribute positions ?
-	}
-
-	if( g instanceof Float32Array || g instanceof Array ) {
-		for( var j = 0; j < g.length; j += 3 ) {
-			var c = j/g.length;
-			this.positions.push( g[ j ], g[ j + 1 ], g[ j + 2 ] );
-			this.positions.push( g[ j ], g[ j + 1 ], g[ j + 2 ] );
-			this.counters.push(c);
-			this.counters.push(c);
+  'use strict'
+
+  var root = this
+
+  var has_require = typeof require !== 'undefined'
+
+  var THREE = root.THREE || (has_require && require('three'))
+  if (!THREE) throw new Error('MeshLine requires three.js')
+
+  function MeshLine() {
+    THREE.BufferGeometry.call(this)
+    this.type = 'MeshLine'
+
+    this.positions = []
+
+    this.previous = []
+    this.next = []
+    this.side = []
+    this.width = []
+    this.indices_array = []
+    this.uvs = []
+    this.counters = []
+    this._points = []
+
+    this.widthCallback = null
+
+    // Used to raycast
+    this.matrixWorld = new THREE.Matrix4()
+
+    Object.defineProperties(this, {
+      // this is now a bufferGeometry
+      // add getter to support previous api
+      geometry: {
+        enumerable: true,
+        get: function() {
+          return this._geometry
+        },
+        set: function(value) {
+          this.setGeometry(value, this.widthCallback)
+        },
+      },
```

Comment:
`geometry` is supposed to be the fallback of the real geometry object. So it was good like before.

For the declarative approach, that might just be another naming as simple as `this._geom` :

```suggestion
      geometry: {
        enumerable: true,
        get: function() {
          return this
        },
      },
      geom: {
        enumerable: true,
        get: function() {
          return this._geom
        },
        set: function(value) {
          this.setGeometry(value, this.widthCallback)
        },
      },
```

The `setGeometry` method is bad, we just keep it as fallback to avoid going to a v2 right now.
So it doesn't need to be perfect.

Reply:
Let's think about how to improve the project for a v2 instead of fighting with this one too much.